pkg/kube/pod/pod_test.go

```diff
@@ -131,3 +133,61 @@ func Test_PodsInNamespaceWithSelectorShouldHaveLabels(t *testing.T) {
 		})
 	}
 }
+
+func TestPodsInNamespaceWithLabelSelectorConvergeToFieldSelector(t *testing.T) {
+	type args struct {
+		kubeClientset kubernetes.Interface
+		expBackoff    wait.Backoff
+		namespace     string
+		labelSelector string
+		fieldSelector string
+	}
+	namespaceName := "test-ns"
+	ns := v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespaceName}}
+	podSucceeded := v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "pod-xhhxj",
+			Namespace: "test-ns",
+			Labels: map[string]string{
+				"app": "test-service",
+			},
+		},
+		Status: v1.PodStatus{
+			Phase: v1.PodSucceeded,
+		},
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "Positive Test",
```

Comment:
Should we add a case where some pods with the label have the field selector set to succeeded and some do not? Should that result in an error in that case.

Reply:
I tried implementing that before marking the PR as ready for review. Listing pods with the fake client returns all the pods added to it, regardless if they have the label-selector/field-selector